cms/djangoapps/contentstore/models.py

```diff
@@ -411,42 +455,12 @@ def _annotate_query_with_ready_to_sync(cls, query_set: QuerySet["EntityLinkBase"
                     Coalesce("upstream_container__publishable_entity__published__version__version_num", 0),
                     Coalesce("version_declined", 0)
                 )
-            )
-        )
-
-    @classmethod
-    def summarize_by_downstream_context(cls, downstream_context_key: CourseKey) -> QuerySet:
-        """
-        Returns a summary of links by upstream context for given downstream_context_key.
-        Example:
-        [
-            {
-                "upstream_context_title": "CS problems 3",
-                "upstream_context_key": "lib:OpenedX:CSPROB3",
-                "ready_to_sync_count": 11,
-                "total_count": 14,
-                "last_published_at": "2025-05-02T20:20:44.989042Z"
-            },
-            {
-                "upstream_context_title": "CS problems 2",
-                "upstream_context_key": "lib:OpenedX:CSPROB2",
-                "ready_to_sync_count": 15,
-                "total_count": 24,
-                "last_published_at": "2025-05-03T21:20:44.989042Z"
-            },
-        ]
-        """
-        result = cls.filter_links(downstream_context_key=downstream_context_key).values(
-            "upstream_context_key",
-            upstream_context_title=F("upstream_container__publishable_entity__learning_package__title"),
-        ).annotate(
-            ready_to_sync_count=Count("id", Q(ready_to_sync=True)),
-            total_count=Count('id'),
-            last_published_at=Max(
-                "upstream_container__publishable_entity__published__publish_log_record__publish_log__published_at"
-            )
+            ),
+            ready_to_sync_from_children=ExpressionWrapper(
+                Exists(subq_container).bitor(Exists(subq_components)),
```

Comment:
We don't really want a bitwise operator here do we? Shouldn't this be
```suggestion
                Exists(subq_container) | (Exists(subq_components),
```
?

Django uses `bitor` for actual bitwise or and the bitwise OR operator `|` for logical OR.

Reply:
Updated: https://github.com/openedx/edx-platform/pull/37208/commits/7751be4bb47c241fba3ce7a08e24fa271dc0b659